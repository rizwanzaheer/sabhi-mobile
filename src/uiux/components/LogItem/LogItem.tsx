import * as React from 'react'
import Container from '../Container/Container'
import Text, { TextTypes } from '../Text/Text'

import { withTheme } from '../../theming'

export enum LogMessageType {
  Info,
  Warning,
  Error,
}

interface LogItemProps {
  /**
   * 0 = info, 1 = Warning/Alert, 2 = Error
   */
  type: LogMessageType

  /**
   * A category for the lo item
   */
  category?: string

  /**
   * A time for the log item
   */
  time: string

  /**
   * Message to be shown
   */
  message: string

  /**
   * Theme passed down from HOC
   */
  theme: any
}

const LogItem: React.FC<LogItemProps> = props => {
  const { WARN, INFO, ERROR } = props.theme.colors.theme
  const logTypes = [WARN, INFO, ERROR]

  return (
    <Container flexDirection={'row'} marginBottom={1} backgroundColor={props.theme.colors.primary.background}>
      <Container w={2} backgroundColor={logTypes[props.type]} />
      <Container flex={1} padding={10}>
        <Container
          flexDirection={'row'}
          justifyContent={'space-between'}
          alignItems={'center'}
          paddingBottom={5}
          paddingRight={true}
        >
          <Text type={TextTypes.SectionHeader}>{props.category}</Text>
          <Text type={TextTypes.SectionHeader}>{props.time}</Text>
        </Container>
        <Container>
          <Text>{props.message}</Text>
        </Container>
      </Container>
    </Container>
  )
}

export default withTheme(LogItem)
