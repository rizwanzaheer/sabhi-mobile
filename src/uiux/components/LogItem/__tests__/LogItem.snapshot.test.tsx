import * as React from 'react'
import { render } from 'react-native-testing-library'
import LogItem from '../LogItem'

describe('Component(snapshots): LogItem', () => {
  it('should render with required props', () => {
    const tree = render(<LogItem message={'Really bad error'} type={0} time={'TODAY'} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render with all props', () => {
    const tree = render(
      <LogItem message={'Really bad error'} category={'The Moon'} type={0} time={'TODAY'} />,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
