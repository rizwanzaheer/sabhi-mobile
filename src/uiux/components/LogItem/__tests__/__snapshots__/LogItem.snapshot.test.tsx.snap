// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshots): LogItem should render with all props 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": "#FFFFFF",
      "borderRadius": undefined,
      "flex": undefined,
      "flexDirection": "row",
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": 1,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": undefined,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": "#ff6666",
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": 2,
      }
    }
  />
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": 1,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": 10,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "space-between",
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": 5,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": 15,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#BBBBBB",
            "flexWrap": "wrap",
            "fontSize": 14,
          }
        }
      >
        The Moon
      </Text>
      <Text
        style={
          Object {
            "color": "#BBBBBB",
            "flexWrap": "wrap",
            "fontSize": 14,
          }
        }
      >
        TODAY
      </Text>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "wrap",
          }
        }
      >
        Really bad error
      </Text>
    </View>
  </View>
</View>
`;

exports[`Component(snapshots): LogItem should render with required props 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": "#FFFFFF",
      "borderRadius": undefined,
      "flex": undefined,
      "flexDirection": "row",
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": 1,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": undefined,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": "#ff6666",
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": 2,
      }
    }
  />
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": 1,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": 10,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "space-between",
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": 5,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": 15,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#BBBBBB",
            "flexWrap": "wrap",
            "fontSize": 14,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#BBBBBB",
            "flexWrap": "wrap",
            "fontSize": 14,
          }
        }
      >
        TODAY
      </Text>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "flexWrap": "wrap",
          }
        }
      >
        Really bad error
      </Text>
    </View>
  </View>
</View>
`;
